metadata-ingestion/src/datahub/ingestion/source/sql/oracle.py

```diff
@@ -147,6 +161,377 @@ def get_table_names(
+            coltype = row[1]
+            length = row[2]
+            precision = row[3]
+            scale = row[4]
+            nullable = row[5] == "Y"
+            default = row[6]
+            comment = row[7]
+            generated = row[8]
+            default_on_nul = row[9]
+            identity_options = row[10]
+
+            if coltype == "NUMBER":
+                if precision is None and scale == 0:
+                    coltype = INTEGER()
+                else:
+                    coltype = ischema_names.get(coltype)(precision, scale)
+            elif coltype == "FLOAT":
+                # TODO: support "precision" here as "binary_precision"
+                coltype = FLOAT()
+            elif coltype in ("VARCHAR2", "NVARCHAR2", "CHAR", "NCHAR"):
+                coltype = ischema_names.get(coltype)(length)
+            elif "WITH TIME ZONE" in coltype:
+                coltype = TIMESTAMP(timezone=True)
+            else:
+                coltype = re.sub(r"\(\d+\)", "", coltype)
+                try:
+                    coltype = ischema_names[coltype]()
+                except KeyError:
+                    warn(
+                        "Did not recognize type '%s' of column '%s'"
+                        % (coltype, colname)
+                    )
+                    coltype = sqltypes.NULLTYPE
+
+            if generated == "YES":
+                computed = dict(sqltext=default)
+                default = None
+            else:
+                computed = None
+
+            if identity_options is not None:
+                identity = self._inspector_instance.dialect._parse_identity_options(  # type: ignore
+                    identity_options, default_on_nul
+                )
+                default = None
+            else:
+                identity = None
+
+            cdict = {
+                "name": colname,
+                "type": coltype,
+                "nullable": nullable,
+                "default": default,
+                "autoincrement": "auto",
+                "comment": comment,
+            }
+            if orig_colname.lower() == orig_colname:
+                cdict["quote"] = True
+            if computed is not None:
+                cdict["computed"] = computed
+            if identity is not None:
+                cdict["identity"] = identity
+
+            columns.append(cdict)
+        return columns
+
+    def get_table_comment(
+        self, table_name: Optional[str], schema: Optional[str] = None
+    ) -> Dict:
+        table_name = self._inspector_instance.dialect.denormalize_name(table_name)
+        schema = self._inspector_instance.dialect.denormalize_name(
+            schema or self.default_schema_name
+        )
+
+        if schema is None:
+            schema = self._inspector_instance.dialect.default_schema_name
+
+        COMMENT_SQL = """
+            SELECT comments
+            FROM dba_tab_comments
+            WHERE table_name = CAST(:table_name AS VARCHAR(128))
```

Comment:
eventually we'll want to replace this with a single bulk fetch for the full schema, instead of fetching each table one at a time

Reply:
I agree with you. This approach (single query to fetch all info in one go) is more preferable. Idea of this PR was different. I can test it and contribute in next PR.